app/lib/features/pins/providers/notifiers/pins_notifiers.dart

```diff
@@ -0,0 +1,144 @@
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/pins/models/pin_edit_state/pin_edit_state.dart';
+import 'package:acter/features/pins/providers/pins_provider.dart';
+import 'package:acter/features/search/providers/pins.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class AsyncPinsNotifier extends AutoDisposeAsyncNotifier<List<ActerPin>> {
+  late Stream<void> _listener;
+
+  Future<List<ActerPin>> _getPins() async {
+    final client = ref.watch(alwaysClientProvider);
+    return (await client.pins()).toList(); // this might throw internally
+  }
+
+  @override
+  Future<List<ActerPin>> build() async {
+    final client = ref.watch(alwaysClientProvider);
+    _listener = client.subscribeStream('PINS'); // stay up to date
+    _listener.forEach((e) async {
+      state = await AsyncValue.guard(() => _getPins());
+    });
+    return _getPins();
+  }
+}
+
+class AsyncPinNotifier
+    extends AutoDisposeFamilyAsyncNotifier<ActerPin, String> {
+  late Stream<void> _listener;
+
+  Future<ActerPin> _getPin() async {
+    final client = ref.watch(alwaysClientProvider);
+    try {
+      return await client.pin(arg);
+    } catch (e) {
+      return await client.waitForPin(arg, null);
+    }
+    // this might throw internally
+  }
+
+  @override
+  Future<ActerPin> build(String arg) async {
+    final client = ref.watch(alwaysClientProvider);
+    _listener = client.subscribeStream(arg); // stay up to date
+    _listener.forEach((e) async {
+      state = await AsyncValue.guard(() => _getPin());
+    });
+    return _getPin();
+  }
+}
+
+class AsyncSpacePinsNotifier
+    extends AutoDisposeFamilyAsyncNotifier<List<ActerPin>, Space> {
+  late Stream<void> _listener;
+
+  Future<List<ActerPin>> _getPins() async {
+    return (await arg.pins()).toList(); // this might throw internally
+  }
+
+  @override
+  Future<List<ActerPin>> build(Space arg) async {
+    final client = ref.watch(alwaysClientProvider);
+    final spaceId = arg.getRoomId();
+    _listener = client.subscribeStream('$spaceId::PINS'); // stay up to date
```

Comment:
yeah, this isn't ever triggered. the string must be `pins` (all lowercase), see https://github.com/acterglobal/a3/blob/f2e223d1c4fda5cc3722c3685e7f0603ea0ebc4a/native/core/src/statics.rs#L18